pkg/commands/git_commands/commit_loader.go

```diff
@@ -495,10 +495,10 @@ func (self *CommitLoader) commitFromPatch(content string) *models.Commit {
 	}
 }
 
-func (self *CommitLoader) setCommitMergedStatuses(ancestor string, commits []*models.Commit) []*models.Commit {
+func setCommitMergedStatuses(ancestor string, commits []*models.Commit) []*models.Commit {
 	passedAncestor := false
 	for i, commit := range commits {
-		if strings.HasPrefix(ancestor, commit.Sha) {
+		if commit.Sha != "" && strings.HasPrefix(ancestor, commit.Sha) {
```

Comment:
Can we add a comment saying 'some commits aren't really commits and don't have sha's, such as the update-ref todo'. I can imagine a reader getting confused about why a commit wouldn't have a sha

Reply:
Sure, added in e61c7c53e81f.